analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SymbolicValueAssertions.cs

```diff
@@ -0,0 +1,164 @@
+                .FailWith("Expected {context:symbolicValue} to have constraints {0}{reason}, but constraints {1} are missing. Actual constraints {2}.",
+                    _ => constraints,
+                    x => x,
+                    _ => Subject.AllConstraints);
+            return new(this);
+        }
+
+        public AndConstraint<SymbolicValueAssertions> HaveOnlyConstraints(params SymbolicConstraint[] constraints)
+            => HaveOnlyConstraints(constraints, string.Empty);
+
+        public AndConstraint<SymbolicValueAssertions> HaveOnlyConstraints(IEnumerable<SymbolicConstraint> constraints, string because = "", params object[] becauseArgs)
+        {
+            HaveConstraintsCommonAssertions(constraints, because, becauseArgs)
+                .Given(allConstraints => new
+                {
+                    missingConstraints = constraints.Except(allConstraints).OrderBy(x => x.ToString()).ToList(),
+                    addionalConstraints = allConstraints.Except(constraints).OrderBy(x => x.ToString()).ToList()
+                })
+                .ForCondition(missingAndAdditional => missingAndAdditional is { missingConstraints.Count: 0 } or { addionalConstraints.Count: > 0 })
+                .FailWith("Expected {context:symbolicValue} to only have constraints {0}{reason}, but constraints {1} are missing. Actual constraints {2}.",
+                    _ => constraints,
+                    missingAndAdditional => missingAndAdditional.missingConstraints,
+                    _ => Subject.AllConstraints)
+                .Then
+                .ForCondition(missingAndAdditional => missingAndAdditional is { missingConstraints.Count: > 0 } or { addionalConstraints.Count: 0 })
+                .FailWith("Expected {context:symbolicValue} to only have constraints {0}{reason}, but {1} additional constraints are present. Actual constraints {2}.",
+                    _ => constraints,
+                    missingAndAdditional => missingAndAdditional.addionalConstraints,
+                    _ => Subject.AllConstraints)
+                .Then
+                .ForCondition(missingAndAdditional => missingAndAdditional is { missingConstraints.Count: 0, addionalConstraints.Count: 0 })
+                .FailWith("Expected {context:symbolicValue} to have constraints {0}{reason}, but constraints {1} are missing and additional constraints {2} are present. Actual constraints {3}.",
+                    _ => constraints,
+                    missingAndAdditional => missingAndAdditional.missingConstraints,
+                    missingAndAdditional => missingAndAdditional.addionalConstraints,
+                    _ => Subject.AllConstraints);
+            return new(this);
+        }
+
+        /// <inheritdoc cref="ObjectAssertions.Be(object, string, object[])"/>
+        public AndConstraint<SymbolicValueAssertions> Be(object expected, string because = "", params object[] becauseArgs)
```

Comment:
This method is a copy of the actual implementation of `ObjectAssertions` (ObjectAssertions derives from ReferenceTypeAssertions). It is called at a couple of places. The call sites should be updated to use one of the methods provided here.